rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1806,4 +1806,36 @@ private void unselectNodes(final HashSet<RMNode> selectedRMNodes) {
         }
     }
 
+    protected void setNodeSources(HashMap<String, NodeSource> hashMap) {
```

Comment:
I see PowerMock is the new name of mockito. I used mockito in the following test to mock the ProActiveRuntimeImpl internal class behaviour. 

https://github.com/ow2-proactive/scheduling/blob/a38bef66235881538ea2541d1ecc9188c5bee6db/rm/rm-server/src/test/java/org/ow2/proactive/resourcemanager/frontend/topology/topologyConcurrencyTest.java

In my test it was used to change the behavior of an existing getter, not to add new methods.  Don't know if it can be used the way you need.

But here you basically add setters, which are meaningful to this class, so I think it's okay.